queries/c/highlights.scm

```diff
@@ -170,6 +172,28 @@
 (preproc_def (preproc_arg) @constant.builtin
   (#any-of? @constant.builtin "stderr" "stdin" "stdout"))
 
+((identifier) @variable.builtin
+  (#any-of? @variable.builtin
+    "__FILE__" "__LINE__" "__DATE__" "__TIME__"
+    "__STDC__" "__STDC_VERSION__" "__STDC_HOSTED__"
+    "__cplusplus" "__OBJC__" "__ASSEMBLER__"
+    "__BASE_FILE__" "__FILE_NAME__" "__INCLUDE_LEVEL__"
+    "__TIMESTAMP__" "__clang__" "__clang_major__"
+    "__clang_minor__" "__clang_patchlevel__"
+    "__clang_version__" "__clang_literal_encoding__"
+    "__clang_wide_literal_encoding__"
+    "__FUNCTION__" "__func__" "__PRETTY_FUNCTION__"
+    "__VA_ARGS__" "__VA_OPT__"))
+((identifier) @variable.builtin
+  (#has-ancestor? @variable.builtin attribute_specifier))
```

Comment:
why not 
```
(attribute_specifier
  (identifier))
```
we should avoid `#has-ancestor?` being invoked on any `(identifier)`. `#has-ancestor?` will call `parent` until it reaches the top query level. Would be ideal if we could remove this predicate.

Reply:
Reaching `__word__` in here is hard:

`typedef int register_t __attribute__((__mode__(__word__)));`

 And gets harder w/ the more nesting. But yeah, I can agree has-ancestor might not be a great idea

Reply:
Realistically, most people don't use compiler intrinsics but the predicate would penalize every single `identifier`.

It would probably be better when the query language would allow such nesting. It could be implemented by letting the outer query enable some state flag when it is matched that could enable matching the inner query. Maybe this could enable implementing nesting during a normal query pass in the Tree-Sitter C runtime. I know the query code is not a pleasure to work with.

Reply:
I was also thinking it'd be better if the query syntax allowed for this as well :)